AttitudeManager/Src/AM.cpp

```diff
@@ -35,13 +35,93 @@ AttitudeManagerInput AttitudeManager::getControlInputs() {
     return temp;
 }
 
+AttitudeManager::AttitudeManager(Flightmode* control_algorithm):
+    control_algorithm_(control_algorithm),
+    motorInstances_(new MotorInstance_t[config::NUM_MOTORS]{})
+{
+    // Go through motor config list and count the # of motors of each axis type
+    uint8_t numYawMotors{0}, numPitchMotors{0}, numRollMotors{0}, numThrottleMotors{0};
+    for (uint8_t i{0}; i < config::NUM_MOTORS; i++) {
+        switch (config::motors[i].axis) {
+            case config::yaw: 
+                numYawMotors++;
+                break;
+            case config::pitch: 
+                numPitchMotors++;
+                break;
+            case config::roll: 
+                numRollMotors++;
+                break;
+        }
+    }
+
+    // Dedicate portions of the motorInstances array to the respective axis by holding references to those positions
+    // We do this instead of creating 4 seperate arrays as we only know the # of total motors and thus don't know the size of the 4 arrays at compile time
+    // If there is 0 of an axis, having the same reference as the next axis will cause it to do nothing
```

Comment:
Yep definitely think it would be easier to allocate 4 arrays instead now. The only thing would be that now we would be allocating arrays based on a computed value instead of just a constant so I don't know if that would make it more unsafe and less preferable